server/src/main/java/org/elasticsearch/search/aggregations/pipeline/AbstractPipelineAggregationBuilder.java

```diff
@@ -102,6 +105,28 @@ public final PipelineAggregator create() throws IOException {
     public void doValidate(AggregatorFactory<?> parent, Collection<AggregationBuilder> factories,
             Collection<PipelineAggregationBuilder> pipelineAggregatorFactories) {
     }
+    
+    /**
+     * Validates pipeline aggregations that need sequentially ordered data.
+     */
+    public static void validateSequentiallyOrderedParentAggs(AggregatorFactory<?> parent, String type, String name) {
+        if (!(parent instanceof HistogramAggregatorFactory || parent instanceof DateHistogramAggregatorFactory
```

Comment:
++ This arrangement is cleaner than an empty `if`, thanks!

One super tiny style nit:  could you change it from a `!(...)` to an `(...) == false`?  We try not to use the negation operator because it's easy to misread, whereas the more verbose `== false` is hard to miss.